tests/30rooms/04messages.pl

```diff
@@ -349,3 +349,55 @@
          })
       });
    };
+
+test "Ephemeral messages received from clients are correctly expired",
+   requires => [ local_user_and_room_fixtures() ],
+
+   do => sub {
+      my ( $user, $room_id ) = @_;
+
+      my $now_ms = int( time() * 1000 );
+      my $filter = '{"types":["m.room.message"]}';
+
+      matrix_send_room_message( $user, $room_id,
+         content => {
+            msgtype                          => "m.text",
+            body                             => "This is a message",
+            "org.matrix.self_destruct_after" => $now_ms + 1000,
```

Comment:
I wonder if we can reduce this and the sleep time, to get the test to run quicker.

Reply:
I'm reducing this to 250ms and the delay to 0.5s.

Reply:
Actually, nevermind, because `--torture-replication` means that if that is too low, when we want to check the message before it expires, the message has already expired by the time it makes it to the `client_reader`...

I've set it to `1000 / 1.5` in the local test and `2000 / 2.5` in the federated tests because those are the values for which I managed to make it work somewhat reliably on my local machine.